src/components/pages/MyAnalyticsPage.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+
+import { Alert, Box, Container, Stack, Typography } from '@mui/material';
+
+import { Loader } from '@graasp/ui';
+
+import { addDays, format, formatISO, intervalToDuration } from 'date-fns';
+import { groupBy } from 'lodash';
+
+import { useAnalyticsTranslation } from '@/config/i18n';
+import { hooks } from '@/config/queryClient';
+import { GroupByInterval } from '@/config/type';
+
+import DateRange from '../common/DateRange';
+import SectionTitle from '../common/SectionTitle';
+import MemberGeneralStatisticsCards from '../custom/MemberGeneralStatisticsCards';
+import ActionsLegend from '../space/charts-layout/ActionsLegend';
+import MemberActionsChart from '../space/charts/MemberActionsChart';
+
+const MyAnalyticsPage = (): JSX.Element => {
+  const { t } = useAnalyticsTranslation();
+
+  const [dateRange, setDateRange] = useState({
+    startDate: addDays(new Date(), -30),
+    endDate: new Date(),
+    key: 'selection',
+  });
+
+  const { data, isLoading } = hooks.useMemberActions({
+    startDate: formatISO(dateRange.startDate),
+    endDate: formatISO(dateRange.endDate),
+  });
+
+  const formattedStartDate = format(dateRange.startDate, 'MMMM d, yyyy');
+  const formattedEndDate = format(dateRange.endDate, 'MMMM d, yyyy');
+
+  const inputValue = `${formattedStartDate} - ${formattedEndDate}`;
```

Comment:
I think this could go in a react's context. Especially if your date picker will affect all charts on this page.